### PR TITLE
[Merged by Bors] - feat: port Topology.Filter

### PR DESCRIPTION
**Pull request by urkud**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Filter.lean [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+import Mathlib.Order.Filter.Lift
+import Mathlib.Topology.Separation
+import Mathlib.Data.Set.Intervals.Monotone
+
+/-!
+# Topology on the set of filters on a type
+
+In this file introduce topology on `Filter α`. It is generated by the sets
+`Set.Iic (𝓟 s) = {l : Filter α | s ∈ l}`, `s : Set α`. A set `s : Set (Filter α)` is open if and
+only if it is a union of a family of these basic open sets, see `Filter.isOpen_iff`.
```
Comment (dupuisf) [inline Mathlib/Topology/Filter.lean:20]:
```suggestion
This file introduces a topology on `Filter α`. It is generated by the sets
`Set.Iic (𝓟 s) = {l : Filter α | s ∈ l}`, `s : Set α`. A set `s : Set (Filter α)` is open if and
only if it is a union of a family of these basic open sets, see `Filter.isOpen_iff`.
```

File: Mathlib/Topology/Filter.lean [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+
+open Filter Topology
+
+variable {ι : Sort _} {α β X Y : Type _}
+
+namespace Filter
+
+/-- Topology on `Filter α` is generated by the sets `Set.Iic (𝓟 s) = {l : Filter α | s ∈ l}`,
+`s : Set α`. A set `s : Set (Filter α)` is open if and only if it is a union of a family of these
+basic open sets, see `Filter.isOpen_iff`. -/
```
Comment (dupuisf) [inline Mathlib/Topology/Filter.lean:50]:
```suggestion
/-- The topology on `Filter α` is generated by the sets `Set.Iic (𝓟 s) = {l : Filter α | s ∈ l}`,
`s : Set α`. A set `s : Set (Filter α)` is open if and only if it is a union of a family of these
basic open sets, see `Filter.isOpen_iff`. -/
```

File: Mathlib/Topology/Filter.lean [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+theorem nhds_inf (l₁ l₂ : Filter α) : 𝓝 (l₁ ⊓ l₂) = 𝓝 l₁ ⊓ 𝓝 l₂ := by
+  simpa only [infᵢ_bool_eq] using nhds_infᵢ fun b => cond b l₁ l₂
+#align filter.nhds_inf Filter.nhds_inf
+
+theorem monotone_nhds : Monotone (𝓝 : Filter α → Filter (Filter α)) :=
+  Monotone.of_map_inf nhds_inf
+#align filter.monotone_nhds Filter.monotone_nhds
+
+theorem Inter_nhds (l : Filter α) : ⋂₀ { s | s ∈ 𝓝 l } = Iic l := by
+  simp_rw [nhds_eq, (· ∘ ·), interₛ_lift'_sets monotone_principal.Iic, Iic, le_principal_iff,
```
Comment (dupuisf) [inline Mathlib/Topology/Filter.lean:151]:
```suggestion
theorem interₛ_nhds (l : Filter α) : ⋂₀ { s | s ∈ 𝓝 l } = Iic l := by
  simp_rw [nhds_eq, (· ∘ ·), interₛ_lift'_sets monotone_principal.Iic, Iic, le_principal_iff,
```

**Review by dupuisf — COMMENTED** [review]
Thanks!

bors d+

**Comment by urkud** [comment]
bors merge

**Pull request closed without merging**